codex/merkletree/merkletree.nim

```diff
@@ -99,76 +139,56 @@ proc leaves*(self: MerkleTree): seq[MerkleHash] =
 proc nodes*(self: MerkleTree): seq[MerkleHash] =
   self.nodes
 
-proc height*(self: MerkleTree): int =
+proc height*(self: MerkleTree): Natural =
   computeTreeHeight(self.leavesCount)
 
-proc `$`*(self: MerkleTree): string =
-  result &= "leavesCount: " & $self.leavesCount
-  result &= "\nnodes: " & $self.nodes
-
-proc getProof*(self: MerkleTree, index: int): ?!MerkleProof =
-  if index >= self.leavesCount or index < 0:
+proc getProof*(self: MerkleTree, index: Natural): ?!MerkleProof =
+  ## Extracts proof from a tree for a given index
+  ## 
+  ## Given a tree built from data blocks A, B and C
+  ##         H5
+  ##      /     \
+  ##    H3       H4
+  ##   /  \     /
+  ## H0    H1 H2
+  ## |     |  |
+  ## A     B  C
+  ##
+  ## Proofs of inclusion (index and path) are
+  ## - 0,[H1, H4] for data block A
+  ## - 1,[H0, H4] for data block B
+  ## - 2,[HZ, H3] for data block C
```

Comment:
Addressing the [comment](https://github.com/codex-storage/nim-codex/pull/504#discussion_r1287730949) by @elcritch  from previous PR.

This is the docs I've added. Let me know if anything is unclear.